File: src/main/java/logan/api/config/BasicConfiguration.java
package logan.api.config;

import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.io.IOException;

public interface BasicConfiguration {
    File getFile();
    YamlConfiguration getConfiguration();
    void setConfiguration(YamlConfiguration configuration);

    default void createKeyIfNoneExists(String key, Object value) {
        YamlConfigurationUtil.setIfNotSet(getConfiguration(), key, value);
    }

    default void createKeyIfNoneExists(String key) {
        createKeyIfNoneExists(key, null);
    }

    default void reload() {
        setConfiguration(YamlConfiguration.loadConfiguration(getFile()));
    }

    default void setValue(String key, Object value) {
        getConfiguration().set(key, value);
    }

    default void save() {
        try {
            getConfiguration().save(getFile());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/logan/api/config/BasicConfiguration.kt
package logan.api.config

import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

interface BasicConfiguration {
    var file: File
    var configuration: YamlConfiguration
    val properties: MutableMap<String, String>

    fun createKeyIfNoneExists(key: String, value: Any? = null) {
        configuration.setIfNotSet(key, value)
    }

    fun reload() {
        configuration = YamlConfiguration.loadConfiguration(file)
    }

    fun save() = configuration.save(file)
}